api/src/Service/UcService.php

```diff
@@ -253,8 +253,88 @@ public function validateUserGroups(array $usergroupIds){
         return $usergroupIds;
     }
 
+    public function createUserGroups(array $result, $type, $update = false) {
+       if ($type == 'Taalhuis' ) {
+           if ($update){
+               $userGroupCoordinator = $this->commonGroundService->getResourceList(['component' => 'uc', 'type' => 'groups'], ['organization' => $result['@id']])['hydra:member'][0];
+               $userGroupEmployee = $this->commonGroundService->getResourceList(['component' => 'uc', 'type' => 'groups'], ['organization' => $result['@id']])['hydra:member'][1];
+           }
+           $coordinator = [
+               'organization' => $result['@id'],
+               'name' => 'TAALHUIS_COORDINATOR',
+               'description' => 'UserGroup coordinator of '.$result['name'],
+           ];
+           if ($update) {
+               $userGroup[] = $this->commonGroundService->updateResource($coordinator,['component' => 'uc', 'type' => 'groups', 'id' => $userGroupCoordinator['id']]);
+           } else {
+               $userGroup[] = $this->commonGroundService->saveResource($coordinator,['component' => 'uc', 'type' => 'groups']);
+           }
+
+           $employee = [
+               'organization' => $result['@id'],
+               'name' => 'TAALHUIS_EMPLOYEE',
+               'description' => 'UserGroup employee of '.$result['name'],
+           ];
+           if ($update) {
+               $userGroup[] = $this->commonGroundService->updateResource($employee,['component' => 'uc', 'type' => 'groups', 'id' => $userGroupEmployee['id']]);
+           } else {
+               $userGroup[] = $this->commonGroundService->saveResource($employee,['component' => 'uc', 'type' => 'groups']);
+           }
```

Comment:
This you can do a little bit more slick (see also next comment for an example): by setting the existing userGroup to null if there is no such thing, you can actually test if the variable "is" (`if($userGroupEmployee)`), instead of needing to have update.